### PR TITLE
Implemented PUT /groups/{groupId} API

### LINKED ISSUE #279: PUT: /groups/{groupId}

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Implemented PUT /groups/{groupId} API

## Review points

Added the controller also

### CODE REVIEW

File: backend/main/src/Shared/Layers/Controller/Controller.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { RepositoryError } from "src/Shared/Layers/Repository/RepositoryError";
 /**
  * This is the controller class.
  * All controller in this should extend this class.
+ * @template T - an input type of this controller.
+ * @template U - a return type of the normal system.
```
Comment (taniguchiiqqq) [inline backend/main/src/Shared/Layers/Controller/Controller.ts:8]: Thanks! I realized this comment was really helpful.

File: backend/main/src/Group/Controllers/UpdateGroupNameController.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	public async handler(request: IUpdateGroupNameUseCase) {
+		const result = await this.updateGroupNameUseCase.execute(request);
+		if (result.err) {
+			if (result.error instanceof UserIsNotAuthorized) {
+				return this.forbidden(result.error.message);
+			}
+			return this.badRequest(result.error);
+		}
+		return this.ok({ groupName: request.name });
```
Comment (taniguchiiqqq) [inline backend/main/src/Group/Controllers/UpdateGroupNameController.ts]: Based on API doc, it should be 
```suggestion
		return this.accepted()
```

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
Thank you for implementation! I mentioned a difference btw implementation and document

File: backend/main/src/Group/Controllers/UpdateGroupNameController.ts (hunk 2 above)
Comment (hitohata) [reply backend/main/src/Group/Controllers/UpdateGroupNameController.ts]: Thank you for pointing it out. It's my mistake. And I've fixed it.

**Review by taniguchiiqqq — APPROVED** [review]
Thanks! LGTM

**Pull request merged**